app/controllers/Checkout.scala

```diff
@@ -1,13 +1,82 @@
 package controllers
 
-import actions.CommonActions.GoogleAuthenticatedStaffAction
+import actions.CommonActions._
+import com.gu.identity.play.PrivateFields
+import model.{AddressData, PaymentData, PersonalData, SubscriptionData}
 import play.api.mvc._
+import services.{IdentityService, CheckoutService}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
 object Checkout extends Controller {
 
-  def renderCheckout = GoogleAuthenticatedStaffAction(Ok(views.html.checkout.payment()))
+  import play.api.data.Forms._
+  import play.api.data._
+
+  private val addressDataMapping = "" -> mapping(
+    "house" -> text,
+    "street" -> text,
+    "town" -> text,
+    "postcode" -> text
+  )(AddressData.apply)(AddressData.unapply)
+
+  private val emailMapping = "" -> tuple(
+    "email" -> email,
+    "confirm" -> email)
+    .verifying("Emails don't match", email => email._1 == email._2)
+    .transform[String](
+      email => email._1, // Transform to a single field
+      email => (email, email) // Reverse transform from a single field to multiple
+    )
+
+  private val personalDataMapping = "" -> mapping(
+    "first" -> text,
+    "last" -> text,
+    emailMapping,
+    addressDataMapping
+  )(PersonalData.apply)(PersonalData.unapply)
+
+  private val paymentDataMapping = "" -> mapping(
+    "account" -> text(8, 8),
+    "sortcode1" -> text(2, 2),
+    "sortcode2" -> text(2, 2),
+    "sortcode3" -> text(2, 2),
+    "holder" -> text
+  )(PaymentData.apply)(PaymentData.unapply)
+
+  val subscriptionForm = Form(mapping(
+    personalDataMapping,
+    paymentDataMapping
+  )(SubscriptionData.apply)(SubscriptionData.unapply))
+
```

Comment:
I know we only have one form but can we move the Form stuff to a separate file? I prefer things to be separated for easier reading, so in this case for the controller I just want to be able to read the actions.